### PR TITLE
Contribute spacemacs doc

### PR DESCRIPTION
**Pull request by binarycrayon**
- add archived Spacemacs.tgz
 - add docset.json
 - specify details in README
 - add icon assets
 - rename Spacemacs.tgz.txt to Spacemacs.tgz

### CODE REVIEW

File: docsets/Spacemacs/docset.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "name": "Spacemacs",
+    "version": "0.200",
+    "archive": "spaceemacs.tgz",
```
Comment (Kapeli) [inline docsets/Spacemacs/docset.json]: Your archive is called `Spacemacs.tgz` instead of `spaceemacs.tgz`.

Comment (binarycrayon) [reply docsets/Spacemacs/docset.json]: sorry typo, fixing

**Comment by Kapeli** [comment]
You still need to fix the difference in case. `spacemacs.tgz != Spacemacs.tgz`

**Comment by binarycrayon** [comment]
@Kapeli Adjusted the last "fix typo" commit to make sure the `Spacemacs.tgz` match both of the file name and docset.json. Will be more rigid in future contributions. Thanks for the patience!

**Comment by binarycrayon** [comment]
checking build

**Pull request closed without merging**